examples/mnist/train_mnist.py

```diff
@@ -1,144 +1,100 @@
-                    default='simple', help='Network type')
-parser.add_argument('--gpu', '-g', default=-1, type=int,
-                    help='GPU ID (negative value indicates CPU)')
-parser.add_argument('--epoch', '-e', default=20, type=int,
-                    help='number of epochs to learn')
-parser.add_argument('--unit', '-u', default=1000, type=int,
-                    help='number of units')
-parser.add_argument('--batchsize', '-b', type=int, default=100,
-                    help='learning minibatch size')
-args = parser.parse_args()
-
-batchsize = args.batchsize
-n_epoch = args.epoch
-n_units = args.unit
-
-print('GPU: {}'.format(args.gpu))
-print('# unit: {}'.format(args.unit))
-print('# Minibatch-size: {}'.format(args.batchsize))
-print('# epoch: {}'.format(args.epoch))
-print('Network type: {}'.format(args.net))
-print('')
-
-# Prepare dataset
-print('load MNIST dataset')
-mnist = data.load_mnist_data()
-mnist['data'] = mnist['data'].astype(np.float32)
-mnist['data'] /= 255
-mnist['target'] = mnist['target'].astype(np.int32)
-
-N = 60000
-x_train, x_test = np.split(mnist['data'],   [N])
-y_train, y_test = np.split(mnist['target'], [N])
-N_test = y_test.size
-
-# Prepare multi-layer perceptron model, defined in net.py
-if args.net == 'simple':
-    model = L.Classifier(net.MnistMLP(784, n_units, 10))
+from chainer import training
+from chainer.training import extensions
+
+
+# Network definition
+class MLP(chainer.Chain):
+
+    def __init__(self, n_in, n_units, n_out):
+        super(MLP, self).__init__(
+            l1=L.Linear(n_in, n_units),  # first layer
+            l2=L.Linear(n_units, n_units),  # second layer
+            l3=L.Linear(n_units, n_out),  # output layer
+        )
+
+    def __call__(self, x):
+        h1 = F.relu(self.l1(x))
+        h2 = F.relu(self.l2(h1))
+        return self.l3(h2)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Chainer example: MNIST')
+    parser.add_argument('--batchsize', '-b', type=int, default=100,
+                        help='Number of images in each mini batch')
+    parser.add_argument('--epoch', '-e', type=int, default=20,
+                        help='Number of sweeps over the dataset to train')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Directory to output the result')
+    parser.add_argument('--resume', '-r', default='',
+                        help='Resume the training from snapshot')
+    parser.add_argument('--unit', '-u', type=int, default=1000,
+                        help='Number of units')
+    args = parser.parse_args()
+
+    print('GPU: {}'.format(args.gpu))
+    print('# unit: {}'.format(args.unit))
+    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('# epoch: {}'.format(args.epoch))
+    print('')
+
+    # Set up a neural network to train
```

Comment:
Should we mention that Classifier reports the loss and accuracy that appear in PrintReport below?